saleor/product/utils.py

```diff
@@ -171,3 +176,18 @@ def price_range_as_dict(price_range):
         return None
     return {'maxPrice': price_as_dict(price_range.max_price),
             'minPrice': price_as_dict(price_range.min_price)}
+
+
+def get_variant_url_from_product(product, attributes):
+    return '%s?%s' % (product.get_absolute_url(), urlencode(attributes))
+
+
+def get_variant_url(variant):
+    attributes = {}
+    variant_attributes = variant.product.product_class.variant_attributes.all()
+
+    for key, value in variant.attributes.items():
+        attribute = variant_attributes.get(pk=key)
+        attributes[attribute.name] = attribute.values.get(pk=value).slug
```

Comment:
Calling `queryset.get()` in a loop will cause a lot of duplicated queries. This should be tested carefully with debug toolbar or some other tool for counting queries.